common/proving_systems.go

```diff
@@ -8,3 +10,26 @@ const (
 	Groth16Bn254
 	SP1
 )
+
+func (t *ProvingSystemId) String() string {
+	return [...]string{"GnarkPlonkBls12_381", "GnarkPlonkBn254", "Groth16Bn254", "SP1"}[*t]
+}
+
+func (t *ProvingSystemId) FromString(provingSystem string) ProvingSystemId {
+	return map[string]ProvingSystemId{
+		"GnarkPlonkBls12_381": GnarkPlonkBls12_381,
+		"GnarkPlonkBn254":     GnarkPlonkBn254,
+		"Groth16Bn254":        Groth16Bn254,
+		"SP1":                 SP1,
+	}[provingSystem]
```

Comment:
I'm wondering if creating the map every time the function is called could be a problem later on. Maybe we can define the map as a global variable with `var` or use `switch`?

Reply:
also, it seems like the reference `t` is not being used, could it be removed?

Reply:
Changed on 9ca7d24706c933d6234160a67d872e6842f40443